securing_apps/topics/cli-apps/cli-apps.adoc

```diff
@@ -0,0 +1,137 @@
+security in your realm.  You can either have one dedicated client whose token will contain all user role mappings
+and claims needed to interact with the REST services your CLI applications invoke on.  Or you can have `kcinit` perform a
+<<_token-exchange,token exchange>> before it passes a token to your CLI application. In that case you'll have
+to set up token exchange permissions.  See our <<_token-exchange,token exchange>> docs for more information.
+
+==== Obtaining Kcinit
+
+The `kcinit` tool is distributed as a native binary.  Binaries for the most
+common operating systems are available from our downloads page.  If a binary
+does not exist for the platform you want to run `kcinit` on, you have the option
+to build it natively directly from link:https://github.com/keycloak/kcinit[sources]
+ using Golang.  See the `kcinit` link:https://github.com/keycloak/kcinit[repo] for more details.
+ 
+==== Configuring Kcinit Locally
+
+If `kcinit` is not configured on the machine it runs on, the end-user
+will be prompted to configure it.
+Installation involves iteractively prompting you for information `kcinit` needs to run.
+
+You can also perform the installation up front by invoking the following command:
+
+----
+     $ kcinit install
+----
+
+This will prompt you for information about the URL of the auth server, the keycloak realm, and the client you want to use to login with.
+This information is stored in the file `$HOME/.keycloak/kcinit/kcinit.yaml`.  If you want to store your configuration someplace else,
+set the `KCINIT_CONFIG` environment variable before running `install`.
+
+==== Usage
+
+After you have installed kcinit, you can login with this command
+
+----
+     $ kcinit login
+----
+
+After successful authentication, `kcinit` stores the login token in
+a file named after the client used to login.  This file will be in the directory `$HOME/.keycloak/kcinit/tokens`.
+
+===== Getting token for your CLI App
+
+The `kcinit token` command will output the access token that was obtained from a login to STDOUT.  You can redirect this
+output to an environment variable or a command line switch for your application.  For example
+
+----
+$ kubectl --token=$(kcinit token)
+----
+
+The `kcinit token` command first looks to see if you have previously performed a login and whether or not
+the token is already stored locally.  It will also check to see if the stored access token needs to be refreshed.
+If you have not logged in yet, or a refresh fails, `kcinit token` will prompt you to login.  All prompt text
+is outputted to STDERR.  After finishing login, the token is stored locally for later use and outputted to STDOUT.
+
+===== Token Exchange and CLI SSO
+
+There are two different ways you can model single sign on if you have multiple CLI applications you are
+securing with `kcinit`.  The first way is that the token created for the dedicated `kcinit` client has all the
```

Comment:
Split into a new paragraph, starting with "The first way..." to parallel the next paragraph that starts with "The second way..."